.zuul.d/dump_zuul_vars.yml

```diff
@@ -0,0 +1,17 @@
+---
+- hosts: controller
+  gather_facts: true
+  tasks:
+    - name: Slurp Zuul inventory
+      slurp:
+        path: "{{ zuul.executor.log_root }}/zuul-info/inventory.yaml"
+      register: _inventory_yaml
+      delegate_to: localhost
+
+    - name: Extract inventory
+      set_fact:
+        _zuul_vars: "{{ _inventory_yaml['content'] | b64decode | from_yaml }}"
+
+    - copy:
+        content: '{{ _zuul_vars.all.vars | to_nice_yaml }}'
+        dest: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/scenarios/centos-9/zuul_inventory.yml'
```

Comment:
You want to add `mode: 0644` else it's using 0600 by default, and may prevent proper read.